File: pascalsTriangle/pascalsTriangle.java
package com.example.java;

import java.util.ArrayList;
import java.util.List;

public class PascalsTriangle {
    public List<List<Integer>> generate(int numRows) {
        List<List<Integer>> triangle = new ArrayList<>();
        while (triangle.size() < numRows) {
            List<Integer> row = new ArrayList<>();
            for (int index = 0; index <= triangle.size(); index++) {
                if (index == 0 || index == triangle.size()) {
                    row.add(1);
                } else if (triangle.size() >= 2) {
                    int leftParent = triangle.get(triangle.size() - 1).get(index - 1);
                    int rightParent = triangle.get(triangle.size() - 1).get(index);
                    row.add(leftParent + rightParent);
                }
            }
            triangle.add(row);
        }
        return triangle;
    }
}
s

File: pascalsTriangle/pascalsTriangle.js
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    // First and last are always 1
    // interior cells are always (index of cell - 1) in previous row + (indexof cell) in previous row
    // ex: row3 cell 2 = row2 cell 1 + row2 cell 2
    var rows = [];
    while (rows.length < numRows) {
        var row = [];
        for (var rowIndex = 0; rowIndex < rows.length + 1; rowIndex++) {
            if (rowIndex === 0 || rowIndex === rows.length) {
                row.push(1);
            } else {
                row.push(rows[rows.length - 1][rowIndex - 1] + rows[rows.length - 1][rowIndex]);
            }
        }
        rows.push(row);
    }
    return rows;
};